modules/compiler/multi_llvm/CMakeLists.txt

```diff
@@ -18,7 +18,9 @@ add_ca_interface_library(multi_llvm)
 
 target_include_directories(multi_llvm INTERFACE
 $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
-target_include_directories(multi_llvm SYSTEM INTERFACE
-  ${LLVM_INCLUDE_DIR})
+if (NOT CA_IN_LLVM_TREE)
```

Comment:
This doesn't feel too good to me. I know you've explained something to us offline but this could do with an explanation in code as there must surely be a better way of handling this. Even if it's a code comment saying it's unclear why it's necessary, or it's a hack, it still alerts future people to maybe question it later on, or not rely on it as unquestionably correct.

Reply:
I've added explanation. I agree that the original design of tagging it into multi_llvm seems wrong, but I think we need to discuss what would be better. This is a relatively small change to how it currently works and we can discuss what should be done going forward.

Reply:
Yep, my gut feeling is that `multi_llvm` looks like it's too much work and shouldn't be acting as the thing that pulls in LLVM headers